src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx

```
@@ -105,9 +114,97 @@ const SelectGitProviderRef = (
   const [connector, setConnector] = useState<ConnectorInfoDTO>()
   const [secret, setSecret] = useState<SecretDTOV2>()
   const scrollRef = useRef<Element>()
+  const oAuthSecretIntercepted = useRef<boolean>(false)
+  const [oAuthStatus, setOAuthStatus] = useState<Status>(Status.TO_DO)
   const { mutate: createSCMConnector } = useCreateDefaultScmConnector({
     queryParams: { accountIdentifier: accountId }
   })
+  const { mutate: createConnector } = useCreateConnector({
+    queryParams: { accountIdentifier: accountId }
+  })
+  const { showError, clear } = useToaster()
+  const [isOAuthSetup, setIsOAuthSetup] = useState<boolean>(false)
+  const PLACEHOLDER = 'placeholder'
+
+  //#region OAuth validation and integration
+  const disableOAuthForGitProvider = useMemo(() => {
+    return gitProvider?.type && [Connectors.GITHUB, Connectors.GITLAB, Connectors.BITBUCKET].includes(gitProvider.type)
+  }, [gitProvider?.type])
+
+  /* Event listener for OAuth server event, this is essential for landing user back to the same tab from where the OAuth started, once it's done */
+  const handleOAuthServerEvent = (event: any): void => {
+    if (!gitProvider) {
+      return
+    }
+    if (event.origin !== getBackendServerUrl() && !isEnvironmentAllowedForOAuth()) {
+      setOAuthStatus(Status.FAILURE)
+      return
+    }
+    if (!event || !event.data) {
+      setOAuthStatus(Status.FAILURE)
+      return
+    }
+    const { accessTokenRef, refreshTokenRef, status, errorMessage } = event.data
+    //safeguard against backend server sending multiple oauth events, which could lead to multiple duplicate connectors getting created
+    if (!oAuthSecretIntercepted.current) {
+      if ((status as string).toLowerCase() === Status.SUCCESS.toLowerCase()) {
+        oAuthSecretIntercepted.current = true
+        createOAuthConnector({ tokenRef: accessTokenRef, refreshTokenRef })
+      } else if (errorMessage !== PLACEHOLDER) {
```

happy with keeping this piece in?

---

Yeah so this is like a temporary solution. These variables `accessTokenRef`, `refreshTokenRef`, `status` and `errorMessage` come with a correct value or as a placeholder value `"placeholder"`. So in this case, I want to make sure only when `errorMessage` equals `"placeholder"`, we mark it as a failure.